### PR TITLE
Fix support for `no_std`

### PR DESCRIPTION
**Pull request by kennykerr**
Rust's `no_std` feature doesn't actually verify that dependencies don't silently add a `std` dependency. It just prevents the current package from taking a direct dependency. This isn't particularly easy to test. 

This PR includes a sequence of commits that will verify success/failure before completion. 

Fixes: #3163 
Fixes: #3164

### CODE REVIEW

**Comment by kennykerr** [comment]
OK, we have a green build. The next commit adds a duplicate panic handler. Duplicate because an indirect `std` dependency will bring in a panic handler and conflict.

**Comment by kennykerr** [comment]
And finally, https://github.com/microsoft/windows-rs/pull/3180/commits/bc4f31374864fad276aee18ede002e14efd20c09 avoids the indirect `std` dependencies and provides the actual fix to the problem. This ended up requiring a bit more work than was originally suggested by #3164.

File: crates/tests/no_std/src/lib.rs [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+}
 
-    // Compilation is sufficient to test.
-    #[test]
-    fn basic() {
-        let object = ComObject::new(App {});
-        drop(object);
-    }
+#[cfg_attr(not(test), panic_handler)]
+fn _panic(_: &core::panic::PanicInfo<'_>) -> ! {
```
Comment (dpaoliello) [inline crates/tests/no_std/src/lib.rs:26]: Can you please add a comment to explain what this is for?

**Review by dpaoliello — DISMISSED** [review]

**Review by dpaoliello — APPROVED** [review]

**Review by ChrisDenton — APPROVED** [review]
Looks like a good solution. An alternative would be to fiddle with `harness = false` in the `Cargo.toml` for the test (to disable the libtest harness) but getting that wired up right might be more effort than it's worth.

**Pull request merged**